### PR TITLE
fix: native selection

### PR DESCRIPTION
**Pull request by thorseraq**
Before:
#527

After:

https://user-images.githubusercontent.com/20554850/211153684-7ee06cea-1a9c-44f2-9ab7-072f7e0ee2dc.mov

### CODE REVIEW

**Comment by himself65** [comment]
What is the bug you fixed? Could you please give a short description for the future people who use git blame

File: packages/blocks/src/__internal__/utils/selection.ts [hunk 1]
```diff
@@ -414,7 +425,16 @@ export function handleNativeRangeDragMove(
   startRange: Range | null,
   e: SelectionEvent
 ) {
-  const isForward = e.x > e.start.x || e.y > e.start.y;
+  const startBlock = getBlockByPoint(e.start);
+  assertExists(startBlock);
+  const startRect = startBlock.getBoundingClientRect();
+
+  const smallDistance = 6;
```
Comment (himself65) [inline packages/blocks/src/__internal__/utils/selection.ts]: why 6?

Comment (thorseraq) [reply packages/blocks/src/__internal__/utils/selection.ts:432]: No particular criterion, any suggestion?

Comment (himself65) [reply packages/blocks/src/__internal__/utils/selection.ts]: nope

Comment (himself65) [reply packages/blocks/src/__internal__/utils/selection.ts]: I think, at least, make all `6` into a constant to declare that they are pointing to the same 6 under same meaning

**Review by doodlewind — APPROVED** [review]

File: packages/blocks/src/__internal__/utils/selection.ts (hunk 1 above)
Comment (thorseraq) [reply packages/blocks/src/__internal__/utils/selection.ts]: Thanks! Done, Let's make it 2px, which is also used in isFarEnough()

**Review by himself65 — APPROVED** [review]

File: packages/blocks/src/__internal__/utils/query.ts [hunk 2]
```diff
@@ -28,6 +29,12 @@ export function getBlockById<T extends ElementTagName>(
   return ele.querySelector<T>(`[${ATTR}="${id}"]` as T);
 }
 
+export function getBlockByPoint(point: IPoint): Element | null | undefined {
+  return document
+    .elementFromPoint(point.x, point.y)
+    ?.closest('[data-block-id]');
```
Comment (lawvs) [inline packages/blocks/src/__internal__/utils/query.ts:35]: you should use `BLOCK_ID_ATTR`

Comment (thorseraq) [reply packages/blocks/src/__internal__/utils/query.ts:35]: ok

**Pull request merged**